src/neo/SmartContract/ApplicationEngine.cs

```diff
@@ -26,13 +26,11 @@ private class InvocationState
         public static event EventHandler<NotifyEventArgs> Notify;
         public static event EventHandler<LogEventArgs> Log;
 
-        public const long GasFree = 0;
```

Comment:
We will never have a free tx?

Reply:
If we have free transactions, then the value of `GasFree` for `Application` and `Verification` is also different. This field is not suitable for being included in `ApplicationEngine`.